[SaferC++] Use smart pointers in the Injected Bundle C API (part 5/5)

zakariaridouh opened the pull request:
#### 1baeb0a85d0f74e0cb256755f2d9b0c324d160a2
<pre>
[SaferC++] Use smart pointers in the Injected Bundle C API (part 5/5)
<a href="https://bugs.webkit.org/show_bug.cgi?id=300297">https://bugs.webkit.org/show_bug.cgi?id=300297</a>
<a href="https://rdar.apple.com/162095972">rdar://162095972</a>

Reviewed by NOBODY (OOPS!).

Final patch to fix violations in the Injected Bundle C API.

Fix them in WKBundleFrame.cpp.

* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleFrame.cpp:
(WKBundleFrameIsMainFrame):
(WKBundleFrameGetParentFrame):
(WKBundleFrameCopyURL):
(WKBundleFrameCopyProvisionalURL):
(WKBundleFrameGetFrameLoadState):
(WKBundleFrameCopyChildFrames):
(WKBundleFrameGetJavaScriptContext):
(WKBundleFrameGetJavaScriptContextForWorld):
(WKBundleFrameGetJavaScriptWrapperForNodeForWorld):
(WKBundleFrameGetJavaScriptWrapperForRangeForWorld):
(WKBundleFrameCopyName):
(WKBundleFrameGetPendingUnloadCount):
(WKBundleFrameGetPage):
(WKBundleFrameStopLoading):
(WKBundleFrameCopyLayerTreeAsText):
(WKBundleFrameAllowsFollowingLink):
(WKBundleFrameGetContentBounds):
(WKBundleFrameGetVisibleContentBounds):
(WKBundleFrameGetVisibleContentBoundsExcludingScrollbars):
(WKBundleFrameGetScrollOffset):
(WKBundleFrameHasHorizontalScrollbar):
(WKBundleFrameHasVerticalScrollbar):
(WKBundleFrameGetDocumentBackgroundColor):
(WKBundleFrameCopySuggestedFilenameForResourceWithURL):
(WKBundleFrameCopyMIMETypeForResourceWithURL):
(WKBundleFrameContainsAnyFormElements):
(WKBundleFrameContainsAnyFormControls):
(WKBundleFrameSetTextDirection):
(WKBundleFrameSetAccessibleName):
(WKBundleFrameCopyWebArchiveFilteringSubframes):
(WKBundleFrameCallShouldCloseOnWebView):
(WKBundleFrameCreateHitTestResult):
(WKBundleFrameCopySecurityOrigin):
(WKBundleFrameFocus):
(_WKBundleFrameGenerateTestReport):
(_WKAccessibilityRootObjectForTesting):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1baeb0a85d0f74e0cb256755f2d9b0c324d160a2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/127644 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/47292 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/38460 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/134915 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/79201 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/2e994268-4b33-4e74-94fe-67b8b009adf2) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47915 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55823 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/97165 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/65197 "") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/c8ceda50-2c7c-402e-8dfb-9dba59f70323) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/130592 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/38318 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/114347 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77646 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/052860a8-cac6-4e70-ba79-3ab98c786b54) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/37122 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/78274 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/108186 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/32891 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/137398 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/54303 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/41857 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/105686 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54814 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/110703 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/105338 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50856 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/29293 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/51924 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/54240 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/60416 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/53475 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/56931 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/55233 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleFrame.cpp:
```diff
@@ -270,21 +270,21 @@ bool WKBundleFrameCallShouldCloseOnWebView(WKBundleFrameRef frameRef)
 WKBundleHitTestResultRef WKBundleFrameCreateHitTestResult(WKBundleFrameRef frameRef, WKPoint point)
```
We should add CF_RETURNS_RETAINED and then the warning should go away in theory.

rniwa commented on Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleFrame.cpp:
```diff
@@ -95,12 +95,12 @@ WKFrameLoadState WKBundleFrameGetFrameLoadState(WKBundleFrameRef frameRef)
 WKArrayRef WKBundleFrameCopyChildFrames(WKBundleFrameRef frameRef)
 {
-    return WebKit::toAPI(&WebKit::toImpl(frameRef)->childFrames().leakRef());    
+    SUPPRESS_UNCOUNTED_ARG return WebKit::toAPI(&WebKit::toProtectedImpl(frameRef)->childFrames().leakRef());
```
We should add CF_RETURNS_RETAINED and then the warning should go away in theory

zakariaridouh committed:
[SaferC++] Use smart pointers in the Injected Bundle C API (part 5/5)
https://bugs.webkit.org/show_bug.cgi?id=300297
rdar://162095972

Reviewed by NOBODY (OOPS!).

Final patch to fix violations in the Injected Bundle C API.

Fix them in WKBundleFrame.cpp.

* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/WebProcess/InjectedBundle/API/c/WKBundleFrame.cpp:
(WKBundleFrameIsMainFrame):
(WKBundleFrameGetParentFrame):
(WKBundleFrameCopyURL):
(WKBundleFrameCopyProvisionalURL):
(WKBundleFrameGetFrameLoadState):
(WKBundleFrameCopyChildFrames):
(WKBundleFrameGetJavaScriptContext):
(WKBundleFrameGetJavaScriptContextForWorld):
(WKBundleFrameGetJavaScriptWrapperForNodeForWorld):
(WKBundleFrameGetJavaScriptWrapperForRangeForWorld):
(WKBundleFrameCopyName):
(WKBundleFrameGetPendingUnloadCount):
(WKBundleFrameGetPage):
(WKBundleFrameStopLoading):
(WKBundleFrameCopyLayerTreeAsText):
(WKBundleFrameAllowsFollowingLink):
(WKBundleFrameGetContentBounds):
(WKBundleFrameGetVisibleContentBounds):
(WKBundleFrameGetVisibleContentBoundsExcludingScrollbars):
(WKBundleFrameGetScrollOffset):
(WKBundleFrameHasHorizontalScrollbar):
(WKBundleFrameHasVerticalScrollbar):
(WKBundleFrameGetDocumentBackgroundColor):
(WKBundleFrameCopySuggestedFilenameForResourceWithURL):
(WKBundleFrameCopyMIMETypeForResourceWithURL):
(WKBundleFrameContainsAnyFormElements):
(WKBundleFrameContainsAnyFormControls):
(WKBundleFrameSetTextDirection):
(WKBundleFrameSetAccessibleName):
(WKBundleFrameCopyWebArchiveFilteringSubframes):
(WKBundleFrameCallShouldCloseOnWebView):
(WKBundleFrameCreateHitTestResult):
(WKBundleFrameCopySecurityOrigin):
(WKBundleFrameFocus):
(_WKBundleFrameGenerateTestReport):
(_WKAccessibilityRootObjectForTesting):
